### PR TITLE
Feature/app insights tracing

### PR DESCRIPTION
**Pull request by ShilovSS**

### CODE REVIEW

File: server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs [hunk 1]
```diff
@@ -54,7 +60,8 @@ protected override async Task RunAsync(CancellationToken cancellationToken)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                ServiceEventSource.Current.ServiceMessage(this.Context, "Request build version");
+                //ServiceEventSource.Current.ServiceMessage(this.Context, "Request build version");
+                this.logger?.LogDebug("Request build version");
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs]: how can it be null?

File: server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs [hunk 2]
```diff
@@ -70,14 +77,14 @@ private async Task UpdateMobileBuild(string mobileType, string buildUrl, string
         {
             try
             {
-                var logStore = new Action<string>(x => ServiceEventSource.Current.ServiceMessage(this.Context, x));
+                var logStore = new Action<string>(x => /*ServiceEventSource.Current.ServiceMessage(this.Context, x)*/this.logger?.LogInformation(x));
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs]: just pass a logger.

File: server2/Arcadia.Assistant/Arcadia.Assistant.MobileBuild/MobileBuildActor.cs [hunk 3]
```diff
@@ -17,35 +18,41 @@ namespace Arcadia.Assistant.MobileBuild
...
+        /// <param name="logger">Logger object interface </param>
+        public MobileBuildActor(ActorService actorService, ActorId actorId, ILogger<MobileBuildActor> logger)
             : base(actorService, actorId)
         {
+            this.logger = logger;
         }
 
         public Task<string> GetMobileBuildVersionAsync(CancellationToken cancellationToken)
         {
+            this.logger?.LogDebug($"Return version");
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.MobileBuild/MobileBuildActor.cs]: how can it be null?

File: server2/Arcadia.Assistant/Arcadia.Assistant.Web/Program.cs [hunk 4]
```diff
@@ -15,13 +20,17 @@ private static void Main()
...
-                // When Service Fabric creates an instance of this service type,
-                // an instance of the class is created in this host process.
+                var configurationPackage = FabricRuntime.GetActivationContext().GetConfigurationPackageObject("Config");
+                var loggerSettings = new LoggerSettings(configurationPackage.Settings.Sections["Logging"]);
 
                 ServiceRuntime.RegisterServiceAsync("Arcadia.Assistant.WebType",
-                    context => new Web(context)).GetAwaiter().GetResult();
+                    context =>
+                    {
+                        var loggerFactory = new LoggerFactoryBuilder(context).CreateLoggerFactory(loggerSettings.ApplicationInsightsKey);
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.Web/Program.cs]: This differs from the way all other SF configuration options are used in Web. Just take AppInsights key the same way, from the configuration object

File: server2/Arcadia.Assistant/Arcadia.Assistant.Web/Startup.cs [hunk 5]
```diff
@@ -128,6 +133,7 @@ public void ConfigureContainer(ContainerBuilder builder)
             builder.RegisterModule(new SickLeavesModule());
             builder.RegisterModule(new PendingActionsModule());
             builder.RegisterModule(new MobileBuildModule());
+            //builder.RegisterServiceLogging(this.AppSettings.Config.Logging);
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.Web/Startup.cs]: changes in this file appear to be junk.

**Review by santee — CHANGES_REQUESTED** [review]

File: server2/Arcadia.Assistant/Arcadia.Assistant.Web/Program.cs (hunk 4 above)
Comment (ShilovSS) [reply server2/Arcadia.Assistant/Arcadia.Assistant.Web/Program.cs]: Missunderstanding is appear. All configuration load in Startup constructor. But its Program class and Startup object hasnt been yet at this moment. 
At the second, class LoggerSettings used for your 'code formatting' promis, of course this single parameter can be readed without class proxing.

File: server2/Arcadia.Assistant/Arcadia.Assistant.Web/Startup.cs (hunk 5 above)
Comment (ShilovSS) [reply server2/Arcadia.Assistant/Arcadia.Assistant.Web/Startup.cs]: Done

File: server2/Arcadia.Assistant/Arcadia.Assistant.MobileBuild/MobileBuildActor.cs (hunk 3 above)
Comment (ShilovSS) [reply server2/Arcadia.Assistant/Arcadia.Assistant.MobileBuild/MobileBuildActor.cs]: Done

File: server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs (hunk 1 above)
Comment (ShilovSS) [reply server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs]: Done

File: server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs (hunk 2 above)
Comment (ShilovSS) [reply server2/Arcadia.Assistant/Arcadia.Assistant.AppCenterBuilds/AppCenterBuilds.cs]: Done

File: server2/Arcadia.Assistant/Arcadia.Assistant.Avatars.Manager/Manager.cs [hunk 6]
```diff
@@ -28,12 +29,14 @@ public class Manager : StatelessService
...
         private readonly IAvatars avatars;
+        private readonly ILogger logger;
 
-        public Manager(StatelessServiceContext context, Func<Owned<CspEmployeeQuery>> employeeQuery, IAvatars avatars)
+        public Manager(StatelessServiceContext context, Func<Owned<CspEmployeeQuery>> employeeQuery, IAvatars avatars, ILogger logger)
             : base(context)
         {
             this.employeeQuery = employeeQuery;
             this.avatars = avatars;
+            this.logger = logger;
```
Comment (santee) [inline server2/Arcadia.Assistant/Arcadia.Assistant.Avatars.Manager/Manager.cs:39]: Error / start logging? Applies to all services.

**Review by santee — APPROVED** [review]

**Pull request merged**